---
title: "Lung Project"
author: "Ali Movasati"
date: "6/30/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Introduction

|   In this report we present the comparison of genomic landscape of lung cohort in two datasets: 1. the PCAWG dataset _ which is a representative of primary tumors; and, 2. the HMF dataset _ which is a representative of metastatic tumors. In total, we analyzed 88 PCAWG lung cancers and 381 HMF lung cancers. These numbers were achieved after removing samples that did not have reliable clinical data available or had dubious sites of origins. Of note, there was no single-cell lung cancer (SCLC) among the PCAWG lung cohort and none of them had smoking records. Therefore, we were not able to subset the PCAWG lung samples into NSCLC/SCLC or non-/current-/former-smoker sub-categories for comparison.


|   In the first part of this analysis, the mutational landscape of two cohorts are summarized and compared. In the second part, mutational signature (MS) contribution for both cohorts were computed with sets of lung-specific SBS, DBS, and ID reference signatures and compared between the datasets.


|   Note that because it was shown that NSCLC and SCLC tumors show significant differences, for a more accurate comparison we will only use **316** NSCLC HMF lung samples. That's because PCAWG cohort only contains this type of lung cancer.

|   All the mean comparison tests between groups in this analysis was done by Wilcoxon test.

## 1. TMB Analysis

|   For both cohorts, the number of single-nucleotide variants (SNVs), multiple-nucleotide variants (MNVs), and small insetions/deletions (INDELs) were obtained from VCF files and their absolute counts were plotted for each sample. In addition, small structural variant (SV) events were also counted and plotted in a separate plot. 




```{r mutation landscape, echo=FALSE, out.width = '100%', message = FALSE, warning=FALSE}


img1_path <- "/home/ali313/Documents/studies/genomic-data-analyst/projects/lung-project/pics-synced/jpg/mut-count-per-sample-pcawg-final-1.jpg"
knitr::include_graphics(img1_path)



img2_path <- "/home/ali313/Documents/studies/genomic-data-analyst/projects/lung-project/pics-synced/jpg/mut-count-per-sample-hmf-final-1.jpg"
knitr::include_graphics(img2_path)


```



Next we plotted the relation of age and TMB (sum of SNVs, MNVs, and INDELs) for each sex and dataset separately. 




```{r tmb and age, echo=FALSE, out.width = '100%', message = FALSE, warning=FALSE}


img1_path <- "/home/ali313/Documents/studies/genomic-data-analyst/projects/lung-project/pics-synced/jpg/age-tmb-final-1.jpg"
knitr::include_graphics(img1_path)


```




|   To compare the number of mutations between the two datasets, we plotted each mutation type separately. Wicoxon test confirms that for small insertions and deletions we see significantly higher frequency in HMF NSCLC samples than in PCAWG NSCLC samples, while for SNVs and MNVs we cannot draw such conclusion.


```{r mutation comparison, echo=FALSE, out.width = '100%', message = FALSE, warning=FALSE}


img1_path <- "/home/ali313/Documents/studies/genomic-data-analyst/projects/lung-project/pics-synced/jpg/tmb-boxplots-final-1.jpg"
knitr::include_graphics(img1_path)


```



|    In addition we visualized number of each mutation type that is annotated as clonal or non-clonal according to the hmf pipeline. Most mutations had unknown clonality, and out of those with a known clonality the larger proportion were clonal mutations. Comparing the clonal to non-clonal between the two datasets, we could not find a significant difference.

```{r clonality, echo=FALSE, out.width = '100%', message = FALSE, warning=FALSE}


img1_path <- "/home/ali313/Documents/studies/genomic-data-analyst/projects/lung-project/pics-synced/jpg/clonality-plot-summary-pcawg-final-1.jpg"
knitr::include_graphics(img1_path)


img2_path <- "/home/ali313/Documents/studies/genomic-data-analyst/projects/lung-project/pics-synced/jpg/clonality-plot-summary-hmf-final-1.jpg"
knitr::include_graphics(img2_path)


img3_path <- "/home/ali313/Documents/studies/genomic-data-analyst/projects/lung-project/pics-synced/jpg/clonality-plot-final-1.jpg"
knitr::include_graphics(img3_path)


```

|   Moreover, the ploidy of genomes of each lung cohort were obtained and compared which also points to no significant difference between the primary and metastatic cancers (p.val < 0.01)


```{r plloidy, echo=FALSE, out.width = '100%', message = FALSE, warning=FALSE}


img1_path <- "/home/ali313/Documents/studies/genomic-data-analyst/projects/lung-project/pics-synced/jpg/ploidy-plot-final-1.jpg"
knitr::include_graphics(img1_path)



```



|   Lastly, the same comparison was done for simple SV events. In this case, the frequencies for all events except deletions with lengths of under 100kb were not different.



```{r sv comparison, echo=FALSE, out.width = '100%', message = FALSE, warning=FALSE}


img1_path <- "/home/ali313/Documents/studies/genomic-data-analyst/projects/lung-project/pics-synced/jpg/sv-boxplots-final-1.jpg"
knitr::include_graphics(img1_path)


```




## 1. MS Analysis

|   Starting with PCAWG dataset, the relative contributions of signatures for each sample was obtained. For SBS, DBS, and INDEL signatures we used 15, 3, and 6 lung-specific reference signatures, respectively. The naming of each signature applies the following pattern: (name of most similar COSMIC reference signature).(COSMIC etiology of that signature).(percentage of cosine similarity between the most similar COSMIC signature and our de-novo extracted signature)_(number of de-novo signature)


### PCAWG Lung Cohort:

Below you can see the relative SBS, DBS and ID signature contribution break-down of each sample: 

* The clustering method that was used was "complete" on all three profiles


```{r pcawg comb sigs, echo=FALSE, out.width = '100%', message = FALSE, warning=FALSE}


# img1_path <- "/home/ali313/Documents/studies/genomic-data-analyst/projects/lung-project/pics-synced/jpg/sbs-rel-cont-stacked-barplot.pcawg-1.jpg"
# knitr::include_graphics(img1_path)
# 
# 
# img2_path <- "/home/ali313/Documents/studies/genomic-data-analyst/projects/lung-project/pics-synced/jpg/dbs-rel-cont-stacked-barplot.pcawg-1.jpg"
# knitr::include_graphics(img2_path)
# 
# 
# img3_path <- "/home/ali313/Documents/studies/genomic-data-analyst/projects/lung-project/pics-synced/jpg/id-rel-cont-stacked-barplot.pcawg-1.jpg"
# knitr::include_graphics(img3_path)


img4_path <- "/home/ali313/Documents/studies/genomic-data-analyst/projects/lung-project/pics-synced/jpg/combined-rel-cont-stacked-barplots.pcawg-1.jpg"
knitr::include_graphics(img4_path)

```


### HMF Lung Cohort:


Below you can see the relative SBS, DBS and ID signature contribution break-down of each sample: 

* The clustering method that was used was "complete" on all three profiles


```{r hmf comb sigs, echo=FALSE, out.width = '100%', message = FALSE, warning=FALSE}



img4_path <- "/home/ali313/Documents/studies/genomic-data-analyst/projects/lung-project/pics-synced/jpg/combined-rel-cont-stacked-barplots-hmf-final-1.jpg"
knitr::include_graphics(img4_path)



```


### MS Comparison between PCAWG and HMF Lung Cohorts:



- For SBS:


In total we used 15 SBS signtures for re-contruction of mutational profiles of samples. Below we compared their absolute and relative contribution between PCAWG and HMF lung cohorts. Here, we can spot several signatures to be significantly different. 



```{r comb sbs sigs (sep), echo=FALSE, out.width = '70%', message = FALSE, warning=FALSE}



img1_path <- "/home/ali313/Documents/studies/genomic-data-analyst/projects/lung-project/pics-synced/jpg/sbs-comparison-absolute-contribution-sep-final-1.jpg"
knitr::include_graphics(img1_path)


img2_path <- "/home/ali313/Documents/studies/genomic-data-analyst/projects/lung-project/pics-synced/jpg/sbs-comparison-relative-contribution-sep-final-1.jpg"
knitr::include_graphics(img2_path)


```


As for several reference mutations we have similar etiologies we also compared their contributions when combined together:

Despite seeing multiple signatures behaving differently we considered separately, when combined, we only see significant differance in Platinum-related signatures.



```{r comb sbs sigs (combined), echo=FALSE, out.width = '70%', message = FALSE, warning=FALSE}



img1_path <- "/home/ali313/Documents/studies/genomic-data-analyst/projects/lung-project/pics-synced/jpg/sbs-comparison-absolute-contribution-comb-final-1.jpg"
knitr::include_graphics(img1_path)


img2_path <- "/home/ali313/Documents/studies/genomic-data-analyst/projects/lung-project/pics-synced/jpg/sbs-comparison-relative-contribution-comb-final-1.jpg"
knitr::include_graphics(img2_path)


```


- For DBS:


Here in absolute contribution plot we only see that DBS signature associated with Platinum is significantly different. However, in relative cotribution plot, tobacco-linked DBS is also significantly different, which logically can also be attributed to the increased level of Platinum mutagenesis which affects all the relative contributions. 

```{r comb dbs sigs, echo=FALSE, out.width = '70%', message = FALSE, warning=FALSE}



img1_path <- "/home/ali313/Documents/studies/genomic-data-analyst/projects/lung-project/pics-synced/jpg/dbs-comparison-absolute-contribution-final-1.jpg"
knitr::include_graphics(img1_path)


img2_path <- "/home/ali313/Documents/studies/genomic-data-analyst/projects/lung-project/pics-synced/jpg/dbs-comparison-relative-contribution-final-1.jpg"
knitr::include_graphics(img2_path)


```



- For ID:


For most ID signatures we see significantly different contributions which is in line with the fact that for small insertions and deletions overall we have higher frequency in HMF compared to PCAWG.

```{r comb id sigs, echo=FALSE, out.width = '100%', message = FALSE, warning=FALSE}



img1_path <- "/home/ali313/Documents/studies/genomic-data-analyst/projects/lung-project/pics-synced/jpg/id-comparison-absolute-contribution-final-1.jpg"
knitr::include_graphics(img1_path)

img2_path <- "/home/ali313/Documents/studies/genomic-data-analyst/projects/lung-project/pics-synced/jpg/id-comparison-relative-contribution-final-1.jpg"
knitr::include_graphics(img2_path)


```













